tests/foreman/installer/test_installer.py

```diff
@@ -1321,6 +1327,197 @@ def extract_help(filter='params'):
                     yield token.replace(',', '')
 
 
+def common_sat_install_assertions(satellite):
+    sat_version = 'stream' if satellite.is_stream else satellite.version
+    assert settings.server.version.release == sat_version
+    result = satellite.execute(
+        r'grep "\[ERROR" --after-context=100 /var/log/foreman-installer/satellite.log'
+    )
+    assert len(result.stdout) == 0
+    result = satellite.cli.Health.check()
+    assert 'FAIL' not in result.stdout
+
+
+def install_satellite(satellite, installer_args):
+    # Register for RHEL8 repos, get Ohsnap repofile, and enable and download satellite
+    satellite.register_to_cdn()
+    satellite.download_repofile(product='satellite', release=settings.server.version.release)
+    satellite.execute('dnf -y module enable satellite:el8 && dnf -y install satellite')
+    # Configure Satellite firewall to open communication
+    satellite.execute(
+        'firewall-cmd --permanent --add-service RH-Satellite-6 && firewall-cmd --reload'
+    )
+    # Install Satellite
+    satellite.execute(
+        InstallerCommand(installer_args=installer_args).get_command(),
+        timeout='30m',
+    )
+    common_sat_install_assertions(satellite)
+
+
+@pytest.fixture(scope='module')
+def sat_default_install(module_sat_ready_rhels):
+    """Install Satellite with default options"""
+    installer_args = [
+        'scenario satellite',
+        f'foreman-initial-admin-password {settings.server.admin_password}',
+    ]
+    install_satellite(module_sat_ready_rhels[0], installer_args)
+    return module_sat_ready_rhels[0]
+
+
+@pytest.fixture(scope='module')
+def sat_non_default_install(module_sat_ready_rhels):
+    """Install Satellite with various options"""
+    installer_args = [
+        'scenario satellite',
+        f'foreman-initial-admin-password {settings.server.admin_password}',
+        'foreman-rails-cache-store type:redis',
+        'foreman-proxy-content-pulpcore-hide-guarded-distributions false',
+    ]
+    install_satellite(module_sat_ready_rhels[1], installer_args)
+    return module_sat_ready_rhels[1]
+
+
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.parametrize(
+    "module_sat_ready_rhels",
+    [get_sat_rhel_version().base_version],
+    indirect=True,
+    ids=[get_sat_rhel_version().major],
```

Comment:
Why you parametrize Satellite RHEL version when we have separated job for each RHEL ?

Reply:
I see,  the fixture is implemented that it always requires parametrization. 
So please fix the `sat_ready_rhel` so it can be called w/o paratemrizing it
```
def get_deploy_args(request):
    deploy_args = {
        'deploy_rhel_version': request.get(param, get_sat_rhel_version().major)
```

Reply:
@lpramuk  updated the pr :)